lightningd/plugin.c

```diff
@@ -201,8 +202,18 @@ void plugin_kill(struct plugin *plugin, char *fmt, ...)
 	kill(plugin->pid, SIGKILL);
 	list_del(&plugin->list);
 
-	if (plugin->start_cmd)
+	/* FIXME: This cleans up as it goes because plugin_kill called twice! */
+	while ((opt = list_top(&plugin->plugin_opts, struct plugin_opt, list))) {
+		if (!opt_unregister(opt->name))
+			fatal("Could not unregister %s from plugin %s",
+			      opt->name, plugin->cmd);
+		list_del_from(&plugin->plugin_opts, &opt->list);
+	}
+
+	if (plugin->start_cmd) {
 		plugin_cmd_killed(plugin->start_cmd, plugin, msg);
+		plugin->start_cmd = NULL;
```

Comment:
Aren't we about to free anyway, which cleans up this dangling pointer?

Reply:
At various points through this series, this is tested in different paths. I think eventually this may be unnecessary, but I'll check